src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -1058,6 +1078,26 @@ public void canPostSynchronousBatch() {
       .forEach(DomainEventAssertions::assertCreateEventForItem);
   }
 
+  @Test
+  public void cannotPostSynchronousBatchUnsafeIfNotAllowed() {
+    JsonArray itemsArray = threeItems();
+    assertThat(postSynchronousBatchUnsafe(itemsArray), statusCodeIs(413));
+  }
+
+  @Test
+  public void canPostSynchronousBatchUnsafe() {
+    OptimisticLockingUtil.configureAllowSuppressOptimisticLocking(
+        Map.of(OptimisticLockingUtil.DB_ALLOW_SUPPRESS_OPTIMISTIC_LOCKING, "9999-12-31T23:59:59Z"));
+
+    JsonArray itemsArray = threeItems();
+    // insert
+    assertThat(postSynchronousBatchUnsafe(itemsArray), statusCodeIs(HttpStatus.HTTP_CREATED));
+    // unsafe update
+    assertThat(postSynchronousBatchUnsafe(itemsArray), statusCodeIs(HttpStatus.HTTP_CREATED));
```

Comment:
What is this update intended to achieve? 

It seems to be a resubmission of the same record representations of from the first request.

This tests doesn't seem to check that the expected state is achieved beyond the status code of the response.

Reply:
If optimistic locking is enabled an insert always succeeds. Therefore we need to do an update to investigate whether optimistic locking is enabled or disabled.

When mod-inventory-storage uses a software library we don't test the software library again but we test whether the software library is correctly used by the mod-inventory-storage code. Here we only need to test the HTTP response code to find out whether the update is rejected with an optimistic locking failure or it succeeded without 200 status code.

If you suspect that the HTTP response code and the record state (unchanged or changed) do not align then this should be tested in the RMB unit tests. We don't want to repeat all these tests for each mod-inventory-storage API (instance, holdings, item) that call the RMB library code for unsafe batch updates.

Reply:
> Why implement this test at all, instead of trusting the existing RMB tests (and the existing tests for the safe batch endpoints)?
I trust that RMB works correctly if called correctly.
But I don't know whether the code in mod-inventory-storage correctly calls the RMB code and whether the response from the RMB code is correctly propagated.
Therefore tests in mod-inventory-storage are needed to show
* that the RMB method is actually called
* that the correct parameters are passed to the RMB method
* that the result from the RMB method is correctly handled, for example returned as JSON body of the mod-inventory-storage API.